### PR TITLE
codeintel: Support bulk deletion of LSIF indexes

### PR DESCRIPTION
**Pull request by chrismwendt**
Resolves https://github.com/sourcegraph/sourcegraph/issues/41196

CleanShot 2022-10-05 at 16 29 11@2x

## Test plan

Go tests, ran locally

### CODE REVIEW

File: internal/codeintel/autoindexing/internal/store/store_indexes.go [hunk 1]
```diff
@@ -167,6 +167,57 @@ JOIN repo ON repo.id = u.repository_id
...
+
+	unset, _ := tx.db.SetLocal(ctx, "codeintel.lsif_indexes_audit.reason", "direct delete by filter criteria request")
+	defer unset(ctx)
+
+	err = tx.db.Exec(ctx, sqlf.Sprintf(deleteIndexesQuery, sqlf.Join(conds, " AND ")))
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (chrismwendt) [inline internal/codeintel/autoindexing/internal/store/store_indexes.go:211]: @efritz Should we do anything else here in addition to deleting the row in `lsif_indexes` (e.g. scheduling a reindex, updating/deleting from the `lsif_uploads` table, etc.)?

File: client/web/src/enterprise/codeintel/uploads/hooks/useDeleteLsifUpload.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ interface UseDeleteLsifUploadResult {
     deleteError: ApolloError | undefined
 }
 
-const DELETE_LSIF_UPLOAD = gql`
+const DELETE_LSIF_UPLOADs = gql`
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/uploads/hooks/useDeleteLsifUpload.tsx:23]: that little `s` looks hilarious 👍

**Review by efritz — APPROVED** [review]
Assuming the TS is good :)

File: internal/codeintel/autoindexing/internal/store/store_indexes.go (hunk 1 above)
Comment (efritz) [reply internal/codeintel/autoindexing/internal/store/store_indexes.go:211]: Nah, if they want to re-index they can mark it without deleting the data so there's no loss of precision (to be implemented). I think delete means delete-without-replacement in this context.

File: client/web/src/enterprise/codeintel/uploads/hooks/useDeleteLsifUpload.tsx (hunk 2 above)
Comment (chrismwendt) [reply client/web/src/enterprise/codeintel/uploads/hooks/useDeleteLsifUpload.tsx]: `DELETE_LSIF_UPLOADₛ` 😬

**Pull request merged**